addons/l10n_in/models/res_config_settings.py

```diff
@@ -15,15 +15,49 @@ class ResConfigSettings(models.TransientModel):
         related="company_id.l10n_in_edi_production_env",
         readonly=False
     )
+    l10n_in_hsn_code_digit = fields.Selection(related='company_id.l10n_in_hsn_code_digit', readonly=False)
+
+    # TDS/TCS settings
+    l10n_in_tds_feature = fields.Boolean(related='company_id.l10n_in_tds_feature', readonly=False)
+    l10n_in_tcs_feature = fields.Boolean(related='company_id.l10n_in_tcs_feature', readonly=False)
+    l10n_in_withholding_account_id = fields.Many2one(related='company_id.l10n_in_withholding_account_id', readonly=False)
+    l10n_in_withholding_journal_id = fields.Many2one(related='company_id.l10n_in_withholding_journal_id', readonly=False)
+    l10n_in_tan = fields.Char(related='company_id.l10n_in_tan', readonly=False)
+
+    # GST settings
+    l10n_in_is_gst_registered = fields.Boolean(related='company_id.l10n_in_is_gst_registered', readonly=False)
+    l10n_in_gstin = fields.Char(string="GST Number", related='company_id.vat', readonly=False)
+    l10n_in_gstin_status_feature = fields.Boolean(related='company_id.l10n_in_gstin_status_feature', readonly=False)
+    l10n_in_gst_efiling_feature = fields.Boolean(related='company_id.l10n_in_gst_efiling_feature', readonly=False)
+    l10n_in_fetch_vendor_edi_feature = fields.Boolean(related='company_id.l10n_in_fetch_vendor_edi_feature', readonly=False)
+    # Automatically set to True if either l10n_in_gst_efiling_feature or l10n_in_fetch_vendor_edi_feature is activated.
+    module_l10n_in_reports = fields.Boolean('GST E-Filing & Matching')
+
+    # E-Invoice
+    l10n_in_edi_feature = fields.Boolean(related='company_id.l10n_in_edi_feature', readonly=False)
+    # Automatically set to True if l10n_in_edi_feature is activated.
     module_l10n_in_edi = fields.Boolean('Indian Electronic Invoicing')
+
+    # E-Waybill
+    l10n_in_ewaybill_feature = fields.Boolean(related='company_id.l10n_in_ewaybill_feature', readonly=False)
+    # Automatically set to True if l10n_in_ewaybill_feature is activated.
     module_l10n_in_ewaybill = fields.Boolean('Indian Electronic Waybill')
-    module_l10n_in_gstin_status = fields.Boolean('Check GST Number Status')
-    module_l10n_in_withholding = fields.Boolean('Indian TDS and TCS')
-    l10n_in_hsn_code_digit = fields.Selection(related='company_id.l10n_in_hsn_code_digit', readonly=False)
-    module_l10n_in_enet_batch_payment = fields.Boolean(string="Vendor Payment")
+
+    # ENet Batch Payment
+    l10n_in_enet_vendor_batch_payment_feature = fields.Boolean(related='company_id.l10n_in_enet_vendor_batch_payment_feature', readonly=False)
+
+    def set_values(self):
```

Comment:
Not possible to make this an inverse on the company field?  (suppose they add the field in the company view)

Reply:
It is actually difficult as in https://github.com/odoo/odoo/blob/f1bb5c4b0247c7b15b3480e90549dce8235a6715/odoo/addons/base/models/res_config.py#L397 we can only install the modules in the end.  However, we can move the feature fields for ewaybill to the module, just have it here with the module_ field and override the res_config_settings_view with a replace changing the module_ field into the feature field.  At the same time, when the module is installed, we make sure that the feature field is set to True for the active company if it is Indian.  (or maybe we could do for all IN companies?  That would correspond to the current behaviour actually)